experimental/devices/st7567/st7567.go

```diff
@@ -0,0 +1,406 @@
+	}
+	return nil
+}
+
+func (d *Display) String() string {
+	switch *d {
+	case DisplayNormal:
+		return "Normal display"
+	case DisplayInverse:
+		return "Inverse display"
+	default:
+		return "Unknown"
+	}
+}
+
+// RegulationResistor is the single regulation resistor value
+type RegulationResistor byte
+
+const (
+	// RegResistorRR0 (0x21): Regulation Resistor ratio
+	RegResistorRR0 RegulationResistor = 0x21
+	// RegResistorRR1 (0x22): Regulation Resistor ratio
+	RegResistorRR1 RegulationResistor = 0x22
+	// RegResistorRR2 (0x24): Regulation Resistor ratio
+	RegResistorRR2 RegulationResistor = 0x24
+)
+
+func (rr *RegulationResistor) Set(s string) error {
+	switch s {
+	case "RR0":
+		*rr = RegResistorRR0
+	case "RR1":
+		*rr = RegResistorRR1
+	case "RR2":
+		*rr = RegResistorRR2
+	default:
+		return fmt.Errorf("unknown RegulataionRatio %q: expected either 'RR0' or 'RR1' or 'RR2'", s)
+	}
+	return nil
+}
+
+func (rr *RegulationResistor) String() string {
+	switch *rr {
+	case RegResistorRR0:
+		return "Regulation resistor RR0"
+	case RegResistorRR1:
+		return "Regulation resistor RR1"
+	case RegResistorRR2:
+		return "Regulation resistor RR2"
+	default:
+		return "Unknown"
+	}
+}
+
+//RegulationRatio selects the regulation resistor ratio
+type RegulationRatio []RegulationResistor
+
+func (rrs *RegulationRatio) String() string {
+	return "regulation resistor ratios"
+}
+
+func (rrs *RegulationRatio) Set(value string) error {
+	values := strings.Split(value, ",")
+
+	for _, v := range values {
+		rr := new(RegulationResistor)
+
+		if err := rr.Set(v); err != nil {
+			return err
+		}
+
+		*rrs = append(*rrs, *rr)
+	}
+
+	return nil
+}
+
+func (rrs RegulationRatio) getValue() RegulationResistor {
+	var out RegulationResistor
+	for _, v := range rrs {
+		out |= v
+	}
+	return out
+}
+
+// Opts contains the configuration for the S7567 device.
+type Opts struct {
+	// Bias selects the LCD bias ratio of the voltage required for driving the LCD.
+	Bias Bias
+	// SegmentDirection is the direction of the segments.
+	SegmentDirection SegmentDirection
+	// CommonDirection is the direction of the segments.
+	CommonDirection CommonDirection
+	// Display changes the selected and non-selected voltage of SEG.
+	Display Display
+	// RegulationRatio controls the regulation ratio of the built-in regulator.
+	RegulationRatio RegulationRatio
+	// StartLine sets the line address of the Display Data RAM to determine the initial display line.
+	StartLine byte
+	// Contrast the value to adjust the display contrast.
+	Contrast byte
+}
+
+// New opens a handle to a ST7567 LCD.
+func New(p spi.Port, dc gpio.PinOut, rst gpio.PinOut, cs gpio.PinIn, o *Opts) (*Dev, error) {
+	c, err := p.Connect(1000*physic.KiloHertz, spi.Mode0, 8)
+
+	if err != nil {
+		return nil, errors.New("could not connect to device")
+	}
+
+	d := &Dev{
+		c:   c,
+		dc:  dc,
+		rst: rst,
+		cs:  cs,
+	}
+
+	cmd := make([]byte, 9)
```

Comment:
Is it intentional that you start with 9 empty bytes?

Reply:
No, this was not intended. I've pushed a new commit (https://github.com/lhooge/periph/commit/fec484da930e58e233160dfe6726070d31166133) without allocating memory first.

Reply:
Please make a new PR since this one was merged.